### ISSUE TITLE
Attaching to an ancestor list feature has a bug?

### ISSUE THREAD

**Issue by diguage** [asciidoctor/asciidoctor#2293]
=== Title

two lines

```
. grandparent list item
.. parent list item
... child list item


+
paragraph attached to grandparent list item
.. parent list item // it should be `list b`, but it is `list a`
. grandparent list item
```

=== Title

one line

```
. grandparent list item
.. parent list item
... child list item

+
paragraph attached to grandparent list item
.. parent list item   // it should be `list b`, but it is `list a`
. grandparent list item
```

=== Title

two lines

```
. grandparent list item
.. parent list item
... child list item


+
paragraph attached to grandparent list item  // should be attach of `grandparent list`
. grandparent list item
```

=== Title

one lines

```
. grandparent list item
.. parent list item
... child list item

+
paragraph attached to grandparent list item // should be attach of `parent list`, but it is attach of `grandparent list`
. grandparent list item
```

**Comment by hleumas** [comment]
I have seen another issue with this:

```
sth::
This

* does
* not

+
work
+
as intended
```

The above will print `+` character before the last paragraph.
<img width="143" alt="screen shot 2017-07-06 at 18 54 11" src="https://user-images.githubusercontent.com/357078/27922796-8ea26b46-627c-11e7-87c2-45d49798193a.png">

**Comment by mojavelinux** [comment]
I am not surprised to discover that there are edge cases with the ancestor list continuation. It's an ambiguous syntax and extremely difficult to get right with respect to how the the list parsing currently works. The only way to really fix it is to rewrite the list parsing code, which is something we have planned.

In the meantime, I strongly recommend using an adjacent continuation chain or open blocks to group non-terminal nested content. In your example, it can be an adjacent chain:

```
sth::
This
+
* does
* not
+
work
+
as intended
```

But if you had a parent item, you might write it as:

```
* parent
+
--
term::
description
+
* item in description
* item in description
--
+
attached to parent
+
also attached to parent
```

In other words, except for rare cases, I would not rely on the ancestor list continuation. It's too fragile.

**Comment by mojavelinux** [comment]
I recognize that nested open blocks are essential to represent more complex structures. Those are planned for 1.5.7.

**Comment by mojavelinux** [comment]
It looks like this has again been reported as #2353.

**Comment by mojavelinux** [comment]
I really don't want to spend any time on this without a full solution to the list parsing problem. Otherwise, we're just wasting effort. We need to switch to single-pass parsing of lists. It's a tough task, but fortunately we have an abundance of tests.

**Comment by 997293** [comment]
Dan,

I'm not sure whether you would consider this as a problem, but one thing of the open block solution is that

````
* level 1
** level 2
+
--
More list levels:

* level 3
** level 4
--
````
In the code above, level 3 and level 4 lists are only preceded by 1 and 2 asterisks. I think this is inconsistent and unintuitive and not that human-readable.

**Comment by mojavelinux** [comment]
You are free to start with any number of asterisks when making lists. So you can write this as follows and it will work exactly the same:

```
* level 1
** level 2
+
--
More list levels:

*** level 3
**** level 4
--
```
